Stack/Opc.Ua.Core/Security/Certificates/CertificateFactory.cs

```diff
@@ -226,7 +226,8 @@ public static X509Certificate2 Load(X509Certificate2 certificate, bool ensurePri
         ushort lifetimeInMonths,
         ushort hashSizeInBits,
         bool isCA,
-        X509Certificate2 issuerCAKeyCert)
+        X509Certificate2 issuerCAKeyCert,
+        byte[] publicKey = null)
```

Comment:
I have a few concerns adding the public key parameter.
I think it were better to have a simple function SignCertificate(
X509Certificate2 certToSign,
X509Certificate2 issuerCAKeyCert
); to handle this case
because all parameters of the cert to sign must stay the same, except IssuerDn. Otherwise all parameters must be passed in.